### PR TITLE
sort src of commutative uop during rewrite

### PR DESCRIPTION
**Pull request by DKormann**
this will sort src of UOp when they are commutative. very helpful for symbolic comparison and should lead to better reusability during rewrite

### CODE REVIEW

**Comment by DKormann** [comment]
this leads to a bug in llvm

**Comment by DKormann** [comment]
solves #5304

File: tinygrad/runtime/ops_llvm.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ def __init__(self, device:LLVMDevice):
     super().__init__("compile_llvm")
   def compile(self, src:str) -> bytes:
     mod = llvm.parse_assembly(src)
+    # print(str(mod))
```
Comment (chenyuxyz) [inline tinygrad/runtime/ops_llvm.py:15]: have you read the pr diff?

**Review by chenyuxyz — COMMENTED** [review]
can you separate the fix (with a regression test) from the main change? it's hard to review a pr with multiple moving parts

File: tinygrad/runtime/ops_llvm.py (hunk 1 above)
Comment (DKormann) [reply tinygrad/runtime/ops_llvm.py:15]: sorry

**Comment by DKormann** [comment]
> can you separate the fix (with a regression test) from the main change? it's hard to review a pr with multiple moving parts

yes will do

**Pull request closed without merging**